src/onepasswordconnectsdk/async_client.py

```diff
@@ -0,0 +1,371 @@
+
+        Raises:
+            FailedToRetrieveItemException: Thrown when a HTTP error is returned
+            from the 1Password Connect API
+
+        Returns:
+            Item object: The found item
+        """
+
+        vault_id = vault
+        if not is_valid_uuid(vault):
+            vault = await self.get_vault_by_title(vault)
+            vault_id = vault.id
+
+        if is_valid_uuid(item):
+            return await self.get_item_by_id(item, vault_id)
+        else:
+            return await self.get_item_by_title(item, vault_id)
+
+    async def get_item_by_id(self, item_id: str, vault_id: str):
+        """Get a specific item by uuid
+
+        Args:
+            item_id (str): The id of the item to be fetched
+            vault_id (str): The id of the vault in which to get the item from
+
+        Raises:
+            FailedToRetrieveItemException: Thrown when a HTTP error is returned
+            from the 1Password Connect API
+
+        Returns:
+            Item object: The found item
+        """
+        url = f"/v1/vaults/{vault_id}/items/{item_id}"
+        response = await self.build_request("GET", url)
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            raise FailedToRetrieveItemException(
+                f"Unable to retrieve item. Received {response.status_code}\
+                     for {url} with message: {response.json().get('message')}"
+            )
+        return self.serializer.deserialize(response.content, "Item")
+
+    async def get_item_by_title(self, title: str, vault_id: str):
+        """Get a specific item by title
+
+        Args:
+            title (str): The title of the item to be fetched
+            vault_id (str): The id of the vault in which to get the item from
+
+        Raises:
+            FailedToRetrieveItemException: Thrown when a HTTP error is returned
+            from the 1Password Connect API
+
+        Returns:
+            Item object: The found item
+        """
+        filter_query = f'title eq "{title}"'
+        url = PathBuilder().vaults(vault_id).items().query("filter", filter_query).build()
+        response = await self.build_request("GET", url)
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            raise FailedToRetrieveItemException(
+                f"Unable to retrieve items. Received {response.status_code} \
+                     for {url} with message: {response.json().get('message')}"
+            )
+
+        if len(response.json()) != 1:
+            raise FailedToRetrieveItemException(
+                f"Found {len(response.json())} items in vault {vault_id} with \
+                    title {title}"
+            )
+
+        item_summary = self.serializer.deserialize(response.content, "list[SummaryItem]")[0]
+        return await self.get_item_by_id(item_summary.id, vault_id)
+
+    async def get_items(self, vault_id: str):
```

Comment:
`[non-blocking]`
I see that the signature for the same function in the sync client is the following (`client.py` - L174):
```py
    def get_items(self, vault_id: str, filter_query: str = None):
```

Should we match the signature here for feature parity? 🤔

Reply:
yep, it should be there. Added.